aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -98,6 +101,80 @@ Tensor _dim_arange(const Tensor& like, int64_t dim) {
   return at::arange(like.size(dim), like.options().dtype(at::kLong));
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~ complex / complex_polar ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TensorIterator complex_op(
+    Tensor& out,
+    const Tensor& a,
+    const Tensor& b,
+    bool check_mem_overlap) {
+  return TensorIteratorConfig()
+      .set_check_mem_overlap(check_mem_overlap)
+      .add_output(out)
+      .add_input(a)
+      .add_input(b)
+      .allow_cpu_scalars(true)
+      .check_all_same_dtype(false)
+      .build();
+}
+
+Tensor& complex_out(Tensor& result, const Tensor& real, const Tensor& imag) {
+  TORCH_CHECK(
+      result.scalar_type() == toComplexType(real.scalar_type()),
+      "Expected object of scalar type ",
+      toComplexType(real.scalar_type()),
+      " but got scalar type ",
+      result.scalar_type(),
+      " for argument 'out'");
+  TORCH_CHECK(
+      real.scalar_type() == imag.scalar_type(),
+      "Expected object of scalar type ",
+      real.scalar_type(),
+      " but got scalar type ",
+      imag.scalar_type(),
+      " for argument 'imag'");
+  auto iter = complex_op(result, real, imag, /*check_mem_overlap=*/true);
+  complex_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor complex(const Tensor& real, const Tensor& imag) {
+  c10::TensorOptions options = real.options();
+  options = options.dtype(toComplexType(real.scalar_type()));
+  Tensor result = at::empty(0, options);
+  return at::complex_out(result, real, imag);
+}
+
+Tensor& complex_polar_out(
+    Tensor& result,
+    const Tensor& abs,
+    const Tensor& angle) {
+  TORCH_CHECK(
```

Comment:
You could add the same checks here. Since you'll be doing the same checks so often you could even write a helper function for them.

Reply:
Sounds good.